client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx

```diff
@@ -247,11 +246,15 @@ const Sidebar = ( { sidebarDomain, siteSlug, submit, goToStep, flow }: SidebarPr
 					taskFilter={ () => enhancedTasks || [] }
 					makeLastTaskPrimaryAction={ true }
 				/>
-				{ showPlansModal && (
+				{ showPlansModal && site?.ID && (
 					<RecurringPaymentsPlanAddEditModal
-						closeDialog={ () => setShowPlansModal( false ) }
-						product={ { subscribe_as_site_subscriber: true } }
-						siteId={ site?.ID }
+						// This spread syntax for props is needed to
+						// resolve an odd IntrinsicAttributes TS error
```

Comment:
It is definitely a work around. Part of the challenge with a real fix is it would require updating the underlying component. However, the underlying component: 
  - Is written in JS, not typescript
  - Is a class component
  - Uses connect() and mapStatetoProps to populate props, particularly siteID (prior to our current usage, I think it was assumed this component would always be used on the admin where we know the site ID). 
  
So there are a lot of odd, complicating issues with resolving it. 

I can do spend the time to resolve this if we like. I'm not sure what all will be involved yet, but it probably adds a fair dose of extra time before merging this.

Reply:
Would a `@ts-ignore` comment work/look better? 🤔  They are rare in the codebase but we do use them in special situations.

Reply:
That would work. I guess the question is if that's better than what I'm already doing? I don't have strong feelings, so I could go any direction. But the @ts-ignore seems reasonable here given the underlying component is not TS.

Reply:
I'm not proficient enough in TS yet to answer confidently. :-) 

I think the "ignore" is more the intent here, and meanwhile we would want to write our code like we normally wouldn't?

Reply:
Yes, that works for me. If you like, I can proceed with that for now to get this merged, but follow up on this in the TypeScript slack channel. If someone has thoughts on a cleaner approach that doesn't involve a major refactor of the underlying component, then handle that in separate PR.

Reply:
Worth noting that adding ts-ignore generates another lint error: `Do not use "@ts-ignore" because it alters compilation errors.eslint[@typescript-eslint/ban-ts-comment](https://typescript-eslint.io/rules/ban-ts-comment)`

Reply:
Check other usages of `@ts-ignore` in calypso, they 're generally:
```js
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
@ts-ignore Explanation why TS was ignored.
```

Reply:
Yup, that's what I did here.